### PR TITLE
Added monogdb TLS option

### PR DESCRIPTION
**Pull request by joda01**
@iegomez I needed TLS support for mongodb login. I added the option hard coded. Probably anyone else needes this option.

### CODE REVIEW

**Review by iegomez — CHANGES_REQUESTED** [review]
Hi! I appreciate the contribution, but it needs a bit of work. It should guide the selection based on a new option to set TLS usage or not and others for any configuration needed, document said options, and also add tests.

**Comment by joda01** [comment]
Okay, I will extend it.

**Comment by joda01** [comment]
@iegomez Added config parameter. Is this correct implemented?

**Comment by iegomez** [comment]
@joda01 Yeah, the option looks correctly set.

File: README.md [hunk 1]
```diff
@@ -1254,6 +1255,7 @@ When not set, these options default to:
 	dbame:           "mosquitto"
 	users: 			 "users"
 	acls:  			 "acls"
+    tls:             "false"
```
Comment (iegomez) [inline README.md:1258]:
```suggestion
    with_tls:             "false"
```
Also, this looks not aligned.

File: backends/mongo.go [hunk 2]
```diff
@@ -95,11 +98,25 @@ func NewMongo(authOpts map[string]string, logLevel log.Level, hasher hashing.Has
...
+
+	if false == m.Tls {
+	  opts = options.ClientOptions {
+	    ConnectTimeout: &to,
+	  }
+	}else{
+	  opts = options.ClientOptions {
+	    ConnectTimeout: &to,
+	    TLSConfig:  &tls.Config{},
+	  }
```
Comment (iegomez) [inline backends/mongo.go:119]:
```suggestion
	opts := options.ClientOptions {
	    ConectTimeout: &to,
	}
	
	if m.Tls {
	    opts.TLSConfig = &tls.Config{}
	}
```

File: backends/mongo.go [hunk 3]
```diff
@@ -95,11 +98,25 @@ func NewMongo(authOpts map[string]string, logLevel log.Level, hasher hashing.Has
...
+	var opts options.ClientOptions
+
+	if false == m.Tls {
+	  opts = options.ClientOptions {
+	    ConnectTimeout: &to,
+	  }
+	}else{
+	  opts = options.ClientOptions {
+	    ConnectTimeout: &to,
+	    TLSConfig:  &tls.Config{},
```
Comment (iegomez) [inline backends/mongo.go:118]: I don't know which options this config might need for Mongo, could you check?

File: backends/mongo.go [hunk 4]
```diff
@@ -29,6 +30,7 @@ type Mongo struct {
 	Conn             *mongo.Client
 	disableSuperuser bool
 	hasher           hashing.HashComparer
+	Tls              bool
```
Comment (iegomez) [inline backends/mongo.go:33]:
```suggestion
	withTLS              bool
```
That would be consistent with how we call the option in `http` and `jwt` backends.

File: README.md [hunk 5]
```diff
@@ -1241,6 +1241,7 @@ auth_opt_mongo_password pwd
 auth_opt_mongo_users users_collection_name
 auth_opt_mongo_acls acls_collection_name
 auth_opt_mongo_disable_superuser true
+auth_opt_mongo_use_tls true
```
Comment (iegomez) [inline README.md:1244]:
```suggestion
auth_opt_mongo_with_tls true
```
That would be consistent with how we call the option in `http` and `jwt` backends.

File: backends/mongo.go (hunk 3 above)
Comment (joda01) [reply backends/mongo.go]: It just needs this option `TLSConfig: &tls.Config{}`. It tells the connector to use TLS connection.
I have tested it with my mongodb and cosmosb mongo db driver.
You can find a full reference at https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo/options#ClientOptions

**Review by joda01 — COMMENTED** [review]
@iegomez fixed your suggestions

**Review by iegomez — APPROVED** [review]
Looks good!

I'd just add the possibility to skip peer verification for testing purposes, see `InsecureSkipVerify` option at https://golang.org/pkg/crypto/tls/#Config. Of course, it should default to `false`, i.e. not skip it, but it's always a useful option for local testing with self signed certificates.

But it's not a blocker and could be added later, I'll leave it to you to decide.

And thanks again for the addition!

**Comment by joda01** [comment]
Good suggestion! I added the skip_verify option. Can you please check again!

Thenks a lot!

**Pull request merged**